encoding/protobq/integration_test.go

```diff
@@ -0,0 +1,81 @@
+package protobq_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/bigquery"
+	"go.einride.tech/protobuf-bigquery/encoding/protobq"
+	publicv1 "go.einride.tech/protobuf-bigquery/internal/examples/proto/gen/einride/bigquery/public/v1"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/testing/protocmp"
+	"gotest.tools/v3/assert"
+)
+
+func Test_Integration_PublicDataSets(t *testing.T) {
+	if _, err := google.FindDefaultCredentials(context.Background()); err != nil {
+		t.Skip("Skipping integration test, missing Google credentials.")
+	}
+	for _, tt := range []*publicDataSetIntegrationTest{
+		{
+			ProjectID: "bigquery-public-data",
+			DatasetID: "san_francisco_film_locations",
+			TableID:   "film_locations",
+			Limit:     100,
+			Message:   &publicv1.FilmLocation{},
+		},
+
+		{
+			ProjectID: "bigquery-public-data",
+			DatasetID: "hacker_news",
+			TableID:   "stories",
+			Limit:     100,
+			Message:   &publicv1.HackerNewsStory{},
+		},
+	} {
+		tt.Run(t)
+	}
+}
+
+type publicDataSetIntegrationTest struct {
+	ProjectID        string
+	DatasetID        string
+	TableID          string
+	Limit            int
+	Message          proto.Message
+	MarshalOptions   protobq.MarshalOptions
```

Comment:
These are unused in the test, I'd like to either add tests with different MarshalOptions or remove the field.

Reply:
These are coming up in an upcoming PR - I have some unsubmitted material that I'm going to portion out after this PR!